### PR TITLE
[go_router] Documentation for StatefulShellRoute

### PR DESCRIPTION
**Pull request by tolo**
Adds documentation about StatefulShellRoute to the Configuration topic (configuration.md).

Fixes flutter/flutter#127209.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by tolo** [comment]
@johnpryan could you have a look at this?

File: packages/go_router/doc/configuration.md [hunk 1]
```diff
@@ -152,6 +152,83 @@ For a complete example, see the [ShellRoute
...
+);
+```
+
+Similar to ShellRoute, a builder must be provided to build the actual shell 
+Widget that encapsulates the branch navigation container. The latter is 
+implemented by the class [StatefulNavigationShell](https://pub.dev/documentation/go_router/latest/go_router/StatefulNavigationShell-class.html), 
+which is passed as the last argument to the builder function. Example:
+
+```dart
+StatefulShellRoute.indexedStack(
```
Comment (chunhtai) [inline packages/go_router/doc/configuration.md:195]: We have example that may be easier to follow https://github.com/flutter/packages/blob/main/packages/go_router/example/lib/others/custom_stateful_shell_route.dart

**Review by chunhtai — COMMENTED** [review]
Thanks for adding the document. Since we start to use code-excerpt, can you try to follow this when adding the example code in doc? https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#add-or-converting-a-code-block

**Comment by chunhtai** [comment]
Hi @tolo are you still working on this pr?

**Comment by tolo** [comment]
> Hi @tolo are you still working on this pr?

Yes, I'll have a look at code-excerpt and update the code / docs.

**Comment by chunhtai** [comment]
closing this pr for now, feel free to reopen when you have time

**Comment by tolo** [comment]
@chunhtai I pushed new commits over a month ago which I don’t think you have reviewed yet (didn’t request a new review though, sorry).

**Review by chunhtai — APPROVED** [review]
LGTM

**Review by hangyujin — APPROVED** [review]
LGTM, thank you for your contribution, can you rebase this to the head to resolve version conflict?

**Pull request merged**